src/commonMain/kotlin/fr/acinq/lightning/crypto/LocalKeyManager.kt

```diff
@@ -50,12 +50,23 @@ data class LocalKeyManager(val seed: ByteVector, val chainHash: ByteVector32) :
         return derivePrivateKey(master, path).privateKey
     }
 
-    override fun bip84Address(account: Long, addressIndex: Long): String {
-        return Bitcoin.computeP2WpkhAddress(bip84PrivateKey(account, addressIndex).publicKey(), chainHash)
+    override fun p2wpkhAddress(keyPath: KeyPath): String {
+        return Bitcoin.computeP2WpkhAddress(privateKey(keyPath).privateKey.publicKey(), chainHash)
+    }
+
+    override fun p2wpkhXpub(keyPath: KeyPath): String {
+        return DeterministicWallet.encode(
+            input = DeterministicWallet.publicKey(privateKey(keyPath)),
+            prefix = when (chainHash) {
+                Block.LivenetGenesisBlock.hash -> DeterministicWallet.zpub
+                Block.TestnetGenesisBlock.hash -> DeterministicWallet.vpub
+                else -> throw IllegalArgumentException("unhandled chain hash $chainHash")
+            }
+        )
     }
 
     override fun closingPubkeyScript(fundingPubKey: PublicKey): Pair<PublicKey, ByteArray> {
-        val priv = bip84PrivateKey(0, 0)
+        val priv = derivePrivateKey(master, finalAddressPath(chainHash)).privateKey
```

Comment:
This change is not necessary but I feel that we should avoid working directly with the `account` and `index` when possible.